### PR TITLE
Move RoleBinding DSPO Role to Notebook Service Account to Server

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1420 

## Description
"edit" user of the project you can still get that binding of the role binding and the service account and can create a workbenches with elyra plugin

## How Has This Been Tested?
Create a DS Project
Create a Workbench with data connection
Give edit permission to regularuser1
check if the regularuser1 can create a workbench with elyra plugin

## Test Impact
Added test cases

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by pnaik1** [comment]
@andrewballantyne , Can you please take a look at this , whenever you find some time?

File: backend/src/routes/api/notebooks/utils.ts [hunk 1]
```diff
@@ -130,3 +138,41 @@ export const enableNotebook = async (
     }
   }
 };
+
+export const getElyraServiceAccountRoleBindingName = (notebookName: string): string =>
```
Comment (andrewballantyne) [inline backend/src/routes/api/notebooks/utils.ts:142]: You duplicated the logic? Why didn't you just move the logic? The frontend code is no longer used now.

File: frontend/src/concepts/pipelines/elyra/__tests__/utils.spec.ts
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/elyra/__tests__/utils.spec.ts:1]: Why did you write a test to verify axios triggered a request? Where are the tests for the utils you moved to the backend? I'm confused at why you didn't use the `__mocks__ ` for notebook?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Please fix the linter issue:
image

---

@pnaik1 we should be talking on slack before posting a PR like this. You don't seem to really have gone after addressing the issue. You copied some stuff to the server, added a odd endpoint and then wrote tests to make sure axios is called in a method that just calls axios?

Also, waiting for me specifically for a review is a poor way to get your code reviewed. I've not been available for real reviews for many months.

**Comment by andrewballantyne** [comment]
I feel this PR should require more conversations before posting. I'm going to close it and we can readdress this effort from Slack.

**Pull request closed without merging**